packages/visual-layer/src/helpers/index.js

```diff
@@ -482,3 +518,44 @@ export const getValidTransformForAggFn = (context) => {
 
 export const getMarkId = (source, schema) => source.filter((val, i) => schema[i] &&
     schema[i].type === FieldType.DIMENSION).join();
+
+export const resolveEncodingValues = (data, i, dataArr, layerInst) => {
+    const transformedValues = {};
+    const values = data.values;
+    const encoding = layerInst.config().encoding;
+    for (const key in values) {
+        const value = getObjProp(encoding[key], 'value');
+        if (value instanceof Function) {
+            transformedValues[key] = value(values, i, dataArr, layerInst);
+        } else {
+            transformedValues[key] = values[key];
+        }
+    }
+    return transformedValues;
+};
+
+export const getColorMetaInf = (color, colorAxis) => ({
+    originalColor: colorAxis.getHslArray(color),
+    stateColor: {},
+    colorTransform: {}
+});
+
+export const toCartesianCoordinates = (points, measurement, rangePlot = false) => {
+    const xOffset = measurement.width / 2;
+    const yOffset = measurement.height / 2;
+    for (let i = 0, len = points.length; i < len; i++) {
+        const point = points[i];
+        const { angle, radius, radius0, angle0 } = point.update;
+        const update = point.update = {
+            x: radius * Math.cos(angle) + xOffset,
+            y: radius * Math.sin(angle) + yOffset
+        };
+        if (rangePlot) {
+            update.x0 = radius * Math.cos(angle) + xOffset;
+            update.y0 = radius * Math.sin(angle) + yOffset;
+            update.x = radius0 * Math.cos(angle0) + xOffset;
+            update.y = radius0 * Math.sin(angle0) + yOffset;
```

Comment:
This line pattern 'radius0 * Math.sin(angle0) + yOffset;' is used 6 times in this function. Can it be used as a function instead?